hibike/hibike_message.py

```diff
@@ -1,55 +1,74 @@
 from __future__ import print_function
 # Rewritten because Python.__version__ != 3
-import serial
 import struct
 import pdb
+import os
+import json
+
+config_file = open(os.path.join(os.path.dirname(__file__), 'hibikeDevices.json'), 'r')
+devices = json.load(config_file)
+#devices = {device["id"]: device for device in devices}
+paramMap = {device["id"]: {param["name"]: (param["number"], param["type"]) for param in device["params"]} for device in devices}
+
+"""
+structure of devices
+{0: 
+    {"id": 0, "name": "LimitSwitch", "params": [
+                                               {"number": 0, "name": "switch0", "type": "bool", "read": True, "write": False}
+                                               ]
+    }
+}
+
+"""
+
+# Dictionary mapping param types to python struct format characters
+paramTypes = {
+  "bool" : "?",
+  "uint8_t": "B",
+  "int8_t": "b", 
+  "uint16_t": "H",
+  "int16_t": "h", 
+  "uint32_t": "I",
+  "int32_t": "i", 
+  "uint64_t": "Q",
+  "int64_t": "q", 
+  "float": "f",
+  "double": "d"
+}
+
 # Dictionary of message types: message id
 messageTypes = {
-  "SubscriptionRequest" :  0x00,
-  "SubscriptionResponse" : 0x01,
-  "DataUpdate" :           0x02,
-  "DeviceUpdate" :         0x03,
-  "DeviceStatus" :         0x04,
-  "DeviceResponse" :       0x05,
-  "Ping" :                 0x06,
-  "DescriptionRequest" :   0x08,
-  "DescriptionResponse" :  0x09,
-  "Error" :                0xff
+  "Ping" :                 0x10,
+  "SubscriptionRequest" :  0x11,
+  "SubscriptionResponse" : 0x12,
+  "DeviceRead" :           0x13,
+  "DeviceWrite" :          0x14,
+  "DeviceData" :           0x15,
+  "Error" :                0xFF
 }
 
 # Dictionary of device types: enumeration
 deviceTypes = {
-  "Limit Switch" :         0x00,
-  "Line Follower" :        0x01,
-  "Potentiometer" :        0x02,
-  "Encoder" :              0x03,
-  "Battery Buzzer" :       0x04,
-  "Team Flag" :            0x05,
-  "Grizzly" :              0x06,
-  "Servo Control" :        0x07,
-  "Linear Actuator" :      0x08
-}
-
-
-# Dictionary of message types: payload length
-messagePayloadLengths = {
-  messageTypes["SubscriptionRequest"] :  2,
-  messageTypes["SubscriptionResponse"] : 13,
-  # TODO: find a clean way to not hardcode this
-  messageTypes["DataUpdate"] :           1,
-  messageTypes["DeviceUpdate"] :         5,
-  messageTypes["DeviceStatus"] :         5,
-  messageTypes["DeviceResponse"] :       5,
-  messageTypes["Error"] :                1
+  "LimitSwitch"   :       0x00,
```

Comment:
Is there a reason these device IDs are inlined into this file, when the JSON could be sourced instead?